etcdctl/ctlv3/command/ep_command.go

```diff
@@ -139,3 +138,29 @@ func epStatusCommandFunc(cmd *cobra.Command, args []string) {
 		os.Exit(ExitError)
 	}
 }
+
+func endpointsFromCluster(cmd *cobra.Command) []string {
+	if !epClusterEndpoints {
+		endpoints, err := cmd.Flags().GetStringSlice("endpoints")
+		if err != nil {
+			ExitWithError(ExitError, err)
+		}
+		return endpoints
+	}
+	c := mustClientFromCmd(cmd)
+	ctx, cancel := commandCtx(cmd)
+	defer func() {
+		c.Close()
+		cancel()
+	}()
+	membs, err := c.MemberList(ctx)
+	if err != nil {
+		ExitWithError(ExitError, err)
```

Comment:
have you checked what the error looks like?

can we provide more context here: "failed to get all endpoints from etcd cluster membership: %v"